From 0c7cac3be12707225d03888a6047e5133d926751 Mon Sep 17 00:00:00 2001
From: Luca Boccassi <luca.boccassi@gmail.com>
Date: Wed, 16 Sep 2020 15:11:17 +0100
Subject: [PATCH] Problem: uninitialized memory access in zarmour_test

Solution: check for size == 0 before dereferencing data pointers

Patch-Source: https://github.com/zeromq/czmq/commit/0c7cac3be12707225d03888a6047e5133d926751
Upstream-Issue: https://github.com/zeromq/czmq/issues/2125
---
 src/zarmour.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/zarmour.c b/src/zarmour.c
index 2860f975c..273eb54cf 100644
--- a/src/zarmour.c
+++ b/src/zarmour.c
@@ -657,8 +657,8 @@ s_armour_decode (zarmour_t *self, const char *test_string, const char *expected,
         zsys_debug ("    decoded '%s' into '%s'", test_string, (char *) zchunk_data (chunk));
     assert (zchunk_size (chunk) == strlen (expected) + 1
             || (strlen (expected) == 0 && zchunk_size (chunk) == 0));
-    assert (streq ((char *) zchunk_data (chunk), expected)
-            || (strlen (expected) == 0 && zchunk_size (chunk) == 0));
+    assert ((strlen (expected) == 0 && zchunk_size (chunk) == 0)
+            || streq ((char *) zchunk_data (chunk), expected));
     zchunk_destroy (&chunk);
 }
 
